src/primitives/sphere.jl

```diff
@@ -37,6 +37,38 @@ end
 
 Sphere(p1::Tuple, p2::Tuple, p3::Tuple) = Sphere(Point(p1), Point(p2), Point(p3))
 
+"""
+    Sphere(p1, p2, p3, p4)
+
+A 3D sphere passing through points `p1`, `p2`, `p3` and `p4`.
+"""
+function Sphere(p1::Point{3}, p2::Point{3}, p3::Point{3}, p4::Point{3})
+  x1 = coordinates(p1)
+  x2 = coordinates(p2)
+  x3 = coordinates(p3)
+  x4 = coordinates(p4)
+  X = hcat(x1, x2, x3, x4)
+  a = 2 * det(vcat(X, ones((1, 4))))
+  T = typeof(a)
+  if isapprox(a, zero(T); atol = atol(T))
+    error("The four points are coplanar.")
+  end
+  q = [x1 ⋅ x1, x2 ⋅ x2, x3 ⋅ x3, x4 ⋅ x4] / a
+  y1 = X[1, :]
+  y2 = X[2, :]
+  y3 = X[3, :]
+  d1 = det(hcat(q, y2, y3, ones(4)))
+  d2 = -det(hcat(q, y1, y3, ones(4)))
+  d3 = det(hcat(q, y1, y2, ones(4)))
+  c = det(hcat(q, y1, y2, y3))
```

Comment:
All these area/volume calculations can be expressed with explicit geometries like area(Triangle(...)). Please try to refactor the code to simplify it and to facilitate future maintenance.

Reply:
Hmm.. I don't know. I think I borrowed this code a long time ago from Asymptote. I don't know what these determinants represent.

Reply:
Ah no i found this code on the [Wolfram site](https://mathworld.wolfram.com/Circumsphere.html).

Reply:
@stla be aware that you cannot copy/paste code from other projects without considering their license terms. If this code comes from Asymptote, it is GPL code, and we can't merge as is.

Reply:
There's a [formula here](http://rodolphe-vaillant.fr/entry/127/find-a-tetrahedron-circumcenter) which looks easier.